vignettes/teal-data-reproducibility.Rmd

```diff
@@ -10,34 +12,50 @@ vignette: >
+It is advisable to always begin analysis in a new session and run all code that pertains to the analysis within the `teal_data` object.
 
-## Verification status
+## Verification
+
+### Verification status
+
+Every `teal_data` object has a _verification status_, which is a statement of whether the contents of the `env` can be reproduced by `code`.
+From this perspective, `teal_data` objects that are instantiated empty are _verified_ but ones instantiated with data and code are _unverified_ because the code need not be reproducible.
+Obviously, `teal_data` objects instantiated with data only are _unverified_ as well.
 
-`teal_data` objects that are instantiated empty are created as _verified_. Objects can be modified only through `eval_code()` and `within()` functions to ensure that the code is saved in the object and always evaluated in the correct environment. Evaluating code in a `teal_data` object _does not_ change its verification status.
+When evaluating code in a `teal_data` object, the code that is stored is the same as the code that is executed, so it is reproducible by definition.
+Therefore, evaluating code in a `teal_data` object _does not_ change its verification status.
+
+The verification status is always printed when inspecting a `teal_data` object.
+Also, when retrieving code, unverified objects add a warning to the code stating that it has not passed verification.
 
 ```{r, message=FALSE, error=TRUE}
 library(teal.data)
 
-my_data <- teal_data()
-my_data <- within(my_data, data <- data.frame(x = 11:20))
-my_data <- within(my_data, data$id <- seq_len(nrow(data)))
-my_data # is verified
-```
+data_empty <- teal_data()
+data_empty # is verified
+data_empty <- within(data_empty, i <- head(iris))
+data_empty # remains verified
 
-Find out more in the `teal_data()` documentation.
+data_with_data <- teal_data(i = head(iris), code = "i <- head(iris)")
+data_with_data # is unverified
+data_with_data <- within(data_with_data, i$rand <- sample(nrow(i)))
+data_with_data # remains unverified
+cat(get_code(data_with_data)) # warning is prepended
+```
 
-## Verification
+### Verification process
 
-`teal_data` objects instantiated with data and code run the risk of the code not being reproducible.
-Such objects will be created in an _unverified_ state. 
-To confirm that the code exactly reproduces the variables stored in the object, one must run the `verify()` function.
-If the verification succeeds, the object's state will be changed to _verified_, otherwise an error will be raised.
-When retrieving code, unverified objects will always add a warning to the code stating that it has not passed verification.
+In order to confirm that the code stored in `teal_data` exactly reproduces the contents of the environment, one must run the `verify()` function.
```

Comment:
What environment?
```suggestion
In order to confirm that the code stored in `teal_data` exactly reproduces the contents of the `env` environment, one must run the `verify()` function.
```

Reply:
Assuming the reader has read the other vignette, they will know. Do you think that is not a valid assumption? Serious question. How many times should we repeat basic intormation?

Reply:
In order to confirm, one _must_. Otherwise confirmation is impossible.

Reply:
I am sorry @chlebowa - I started reading this vignette and then realized we have the second one with basic information. When starting only with this one this was not clear, but no one will start just with this one, but will go through the other one first. So ignore please :) resolving